dask/array/rechunk.py

```diff
@@ -209,6 +209,24 @@ def blockshape_dict_to_tuple(old_chunks, d):
 DEFAULT_BLOCK_SIZE_LIMIT = 1e8
 
 
+def rechunk_preserve_number(x, *args, **kwargs):
+    """Make new regular chunks with the same number of chunks on each dim"""
+    out = []
+    for size, ch in zip(x.shape, x.chunks):
+        if (len(set(ch[:-1])) > 1) or (ch[-1] > ch[0]):
+            # irregular
+            l = len(ch)
+            newsize = int(np.ceil(size / l))
+            out.append(tuple([newsize] * (l - 1) + [size - newsize * (l - 1)]))
+        else:
+            # already regular
+            out.append(ch)
+    return tuple(out)
+
+
+rechunkers = {'preserve_number': rechunk_preserve_number}
```

Comment:
Do you have other ideas for possible "rechunkers"?

I'm not sure that this rechunking strategy is common or non-trivial enough that it is worth adding directly to `rechunk()`. Most of the time users know the chunks they want.

For a utility function like `to_zarr()`, I might write something even simpler, like `arr.rechunk([c[0] for c in arr.chunks])`. Or you could even write this function in one long line, with `arr.rechunk([int(np.ceil(s / len(c))) for s, c in zip(arr.shape, arr.chunks)])`.

Reply:
See ideas here: https://github.com/dask/dask/issues/3302#issuecomment-385674820

Reply:
Any thoughts on shorter names?

Reply:
I wanted these to be fairly descriptive. They could be shorter if the description of what they do does into the docstring of `rechunk`.